File: Java/src/test/java/org/revian/MyQueueTest.java
package org.revian;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class MyQueueTest {
    private MyQueue queue;

    @BeforeEach
    void setUp() {
        queue = new MyQueue();
    }

    @Test
    void testEnqueueAndDequeue() {
        queue.enqueue(1);
        queue.enqueue(2);
        assertEquals(1, queue.dequeue());
        assertEquals(2, queue.dequeue());
    }

    @Test
    void testSize() {
        queue.enqueue(1);
        queue.enqueue(2);
        assertEquals(2, queue.size());
    }

    @Test
    void testDequeueEmptyQueue() {
        assertThrows(IllegalStateException.class, () -> queue.dequeue());
    }
}

File: Scala/src/test/scala/org/revian/MyQueueTest.scala
package org.revian

import org.scalatest.funsuite.AnyFunSuite

class MyQueueTest extends AnyFunSuite {

  test("Enqueue element into MyQueue") {
    val queue = new MyQueue(3)
    queue.enqueue(1)
    assert(queue.size() == 1)
  }

  test("Dequeue element from MyQueue") {
    val queue = new MyQueue(3)
    queue.enqueue(1)
    queue.enqueue(2)
    assert(queue.dequeue() == 1)
    assert(queue.size() == 1)
  }

  test("Dequeue from empty MyQueue should throw exception") {
    val queue = new MyQueue(3)
    assertThrows[IllegalStateException] {
      queue.dequeue()
    }
  }

  test("Enqueue beyond capacity should throw exception") {
    val queue = new MyQueue(3)
    queue.enqueue(1)
    queue.enqueue(2)
    queue.enqueue(3)
    assertThrows[IllegalStateException] {
      queue.enqueue(4)
    }
  }

  test("Circular behavior - wrap around") {
    val queue = new MyQueue(3)
    queue.enqueue(1)
    queue.enqueue(2)
    queue.dequeue()
    queue.enqueue(3)
    assert(queue.size() == 2)
    assert(queue.dequeue() == 2)
    assert(queue.dequeue() == 3)
  }
}
